### PR TITLE
docs: Clarify FIFO Behavior of tokio::sync::Mutex

### PR DESCRIPTION
**Pull request by azuredream**
Closes: #6278

### CODE REVIEW

File: tokio/src/sync/mutex.rs [hunk 1]
```diff
@@ -17,6 +17,11 @@ use std::{fmt, mem, ptr};
 /// differences: [`lock`] is an async method so does not block, and the lock
 /// guard is designed to be held across `.await` points.
 ///
+/// Please note that Tokio's Mutex operates on a guaranteed FIFO basis.
```
Comment (Darksonn) [inline tokio/src/sync/mutex.rs:20]: This is redundant.
```suggestion
/// Tokio's Mutex operates on a guaranteed FIFO basis.
```

File: tokio/src/sync/mutex.rs [hunk 2]
```diff
@@ -17,6 +17,11 @@ use std::{fmt, mem, ptr};
...
+/// This means that the order in which tasks call the [`lock`] method is
+/// the exact order in which they will acquire the lock.
+/// We commit to maintaining this behavior in future versions.
```
Comment (Darksonn) [inline tokio/src/sync/mutex.rs:23]: This is generally implicit, so it's redundant to include it.
```suggestion
```

**Review by Darksonn — COMMENTED** [review]
Thanks, a few suggestions.

**Comment by azuredream** [comment]
Updated. Thanks for your review!

**Review by Darksonn — APPROVED** [review]
Thank you.

**Pull request merged**